# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpSender.h
# Commit: 99ea4f221834
# Full Hash: 99ea4f221834f706be3738a92183287b54d8f718
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-22 17:00:32
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Make RTCRtpSender::GetStats get ICE stats. r=ng
#   
#   Also fix bugs where we would get duplicate ICE stats. This mostly mattered to
#   WebrtcGlobalInformation since RTCStatsReportInternal does not automatically
#   remove duplicate ids like RTCStatsReport (RTCStatsReport is maplike).
# ==============================================================================

diff -r 5801060dd3d0 -r 99ea4f221834 dom/media/webrtc/jsapi/RTCRtpSender.h
--- a/dom/media/webrtc/jsapi/RTCRtpSender.h	Thu Dec 22 02:33:56 2022 +0000
+++ b/dom/media/webrtc/jsapi/RTCRtpSender.h	Thu Dec 22 02:33:57 2022 +0000
@@ -71,7 +71,8 @@
       ErrorResult& aRv);
 
   nsPIDOMWindowInner* GetParentObject() const;
-  nsTArray<RefPtr<RTCStatsPromise>> GetStatsInternal();
+  nsTArray<RefPtr<RTCStatsPromise>> GetStatsInternal(
+      bool aSkipIceStats = false);
 
   // This would just be stream ids, except PeerConnection.jsm uses GetStreams
   // to implement the non-standard RTCPeerConnection.getLocalStreams. We might
@@ -162,6 +163,7 @@
   bool mSimulcastEnvelopeSet = false;
   Maybe<RTCRtpSendParameters> mLastReturnedParameters;
   RefPtr<MediaPipelineTransmit> mPipeline;
+  RefPtr<MediaTransportHandler> mTransportHandler;
   RefPtr<RTCRtpTransceiver> mTransceiver;
   nsTArray<RefPtr<DOMMediaStream>> mStreams;
   bool mHaveSetupTransport = false;
